From 7d77117dd906970c5b3527cc078446a6674128a4 Mon Sep 17 00:00:00 2001
From: David Mohammed <fossfreedom@ubuntu.com>
Date: Sun, 19 Feb 2017 22:46:50 +0000
Subject: [PATCH] check that schema and prefix have been previously created -
 when removing C based widgets the private objects are deleted before applet
 settings are interrogated
Forwarded: https://github.com/budgie-desktop/budgie-desktop/pull/805
Origin: Other

---
 src/plugin/applet.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugin/applet.c b/src/plugin/applet.c
index e600ba1..a2c92fe 100644
--- a/src/plugin/applet.c
+++ b/src/plugin/applet.c
@@ -231,7 +231,7 @@ GSettings *budgie_applet_get_applet_settings(BudgieApplet *self, gchar *uuid)
         GSettings *settings = NULL;
         gchar *path = NULL;
 
-        if (!self->priv->schema || !self->priv->prefix) {
+        if (!self || !self->priv || !self->priv->schema || !self->priv->prefix) {
                 return NULL;
         }
 
-- 
2.7.4

